components/camel-spring-boot/src/main/java/org/apache/camel/spring/boot/RoutesCollector.java

```diff
@@ -84,7 +84,8 @@ public void onApplicationEvent(ContextRefreshedEvent event) {
         CamelContext camelContext = applicationContext.getBean(CamelContext.class);
 
         // only add and start Camel if its stopped (initial state)
-        if (camelContext.getStatus().isStopped()) {
+        if (((ContextRefreshedEvent) event).getApplicationContext().getId().equals(this.applicationContext.getId())
```

Comment:
According to [the documentation](https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/context/ApplicationContext.html#getId--) `ApplicationContext::getId` can return `null`, this would result in `NullPointerException` thrown from this line.

`event` is already a `ContextRefreshedEvent` no need to cast to it.

```suggestion
        if (event.getApplicationContext() == this.applicationContext)
```